### ISSUE TITLE
Console error when data-simplebar applied to dynamically generated element

### ISSUE THREAD

**Issue by hypersurface** [Grsmto/simplebar#340]
## Current Behavior
I'm using AngularJS 1.5 and Angular Material 1.1.14. Adding data-simplebar seems to work fine except when applied to a div inside the material `md-sidenav` directive. I get the following error in the browser console: 

> angular.js:14195 TypeError: Cannot read property 'offsetHeight' of undefined at SimpleBar.recalculate (simplebar.js:4167)

I suspect this is because the `md-sidenav` is dynamically generated. The error doesn't show up until I click to view the sidenav and then hover over the scrollable area.

I also sometimes see this error:

> Error: this.heightAutoObserverEl is undefined

## Suggested solution(s)
I wrapped the `_proto.recalculate()` code inside `if (this.heightAutoObserverEl !== undefined) {...}` and the console error no longer appears. Also, simplebar still seems to work fine. Honestly, not sure if this is the best solution though.

## Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| SimpleBar        | 4.0
| Browser          | Chrome, Firefox
| npm/Yarn         |
| Operating System | Mac OS 10.14.5

**Comment by kubami9** [comment]
Had the same problem without Angular and I think I found solution

- SimpleBar 4.2.3
- Brave (Chrome) and Firefox
- npm
- Ubuntu 18.04.3 LTS

**Issue description:**
I have content in a div with position absolute and it's in a wrapper with resizable functionality on which I apply simplebar. Inside the content div I have accordions. When resizing the wrapper simplebar recalculates and everything goes right but when switching accordions in child component it makes content taller but simplebar can't have any clue about it so I recalculate manually and get such error
`TypeError: Cannot read property 'offsetHeight' of undefined
    at SimpleBar.recalculate (simplebar.esm.js:533)
    at HTMLDivElement.toggleAccordion (init.js:63)` 

**Solution:**
**TL;DR remove `data-simplebar` attribute from your HTML element. (I think it's also worth pointing out in the docs).**

When you write:
`const simpleBar = new SimpleBar(document.querySelector('.someElement'));`
in JS it creates new instance of SimpleBar for this element. If you don't have previously mentioned attribute on your HTML element it's initialized with init function containing `initDOM` function which does all the stuff including creating and appending `this.heightAutoObserverEl` which is actually a div and everything works fine. If you have that attribute however, element is found by it before getting to your JS declaration of `simpleBar` and `initDOM` is fired on it so `this.heightAutoObserverEl` exists only there. Then when everything is loaded `initDOM` doesn't fire anymore on your instance and therefor you're left with new instance which doesn't have `this.heightAutoObserverEl` so your code fails in the 1st line of `recalculate` function. The fact it partially works (in my case) is that when resizing component it uses old intance of `recalculate` in the context of first SimpleBar.

This is my guess and it works but if the reason turns out to be different I'm curious to know about it.

**Comment by kubami9** [comment]
Actually manual setting is mentioned as "Other usage" so I guess it's pretty clear data attribute is not needed then :man_shrugging:

**Comment by Grsmto** [comment]
Hi @kubami9 thanks for your analyse and debug.
>but simplebar can't have any clue about it so I recalculate manually 
This should not be the case, SimpleBar does know about any child content being resized and should recalculate automatically. If everything works, you should never ever have to call `recalculate`. I think there is something specific to your setup that breaks SimpleBar and so you get that error.

From the setup you describe, I'm wondering if it's not because the element you're applying SimpleBar to is not hidden during initialisation. But I'm not sure, it needs to be replicated.

If you have time and want to help on this, that would be extremely useful if you could do a replicate/isolated example of this problem using the [JSFiddle example](https://jsfiddle.net/Grsmto/57obe5w2/1/) or [Codesandbox](https://codesandbox.io/s/simplebar-react-gwgyw)

**Comment by kubami9** [comment]
Hello @Grsmto 

Here is replicate example of the problem: https://jsfiddle.net/eg1cywvs/43/

So the actual element that is being hidden/shown is a child's child. If it's initially shown scrolling works but as you hide it it's not recognized and scroll is still there.

**Comment by kubami9** [comment]
Isn't it caused by the fact child element has `position: absolute` so it's removed from normal document flow? After removing it everything works fine.

**Comment by Grsmto** [comment]
Actually yes if your inner element has `position: absolute` it can't trigger an overflow on its parent. This is not a SimpleBar issue but just pure CSS I believe. 
Also for angular please have a try with the `simplebar-angular` wrapper.
So I'm gonna close this issue!
Thanks!
